frameworks/modular/odp_module.h

```diff
@@ -237,6 +228,32 @@ done:								\
 	return result;						\
 }
 
+/* ODP_SUBSYSTEM_FOREACH_TEMPLATE() generates subsystems
+ * initialization and termination routines automatically,
+ * examples:
+ *
+ * ODP_SUBSYSTEM_FOREACH_TEMPLATE(subsystem, init_global, DBG)
+ * will generate a function walk through all the modules of the
+ * subsystem and invoke init_global method for each.
+ */
+
+#define ODP_SUBSYSTEM_FOREACH_TEMPLATE(subs, method, print)	\
+int odp_ ## subs ##_## method(bool continue_on_errors)		\
+_SUBSYSTEM_FOREACH_TEMPLATE(subs, method, print)
+
+/* SUBSYSTEM_FOREACH_TEMPLATE() generates static walk through
+ * functions to invoke subsystem's modules initialization or
+ * termination methods, examples:
+ *
+ * SUBSYSTEM_FOREACH_TEMPLATE(subsystem, init_global, DBG)
+ * will generate a function walk through all the modules of the
+ * subsystem and invoke init_global method for each.
+ */
+
+#define SUBSYSTEM_FOREACH_TEMPLATE(subs, method, print)	\
```

Comment:
seems no need to add an extra macro here?

ODP_SUBSYSTEM_FOREACH_TEMPLATE(...) is extern
we can just use to generate a static function use the same macro:
static ODP_SUBSYSTEM_FOREACH_TEMPLATE(...)